tests/deep_inspection/test_standalone_di.py

```
@@ -49,3 +70,202 @@ def test_wait_for_completion(self) -> None:
+                    return
+        except TimeoutExpiredError:
+            remaining = list(
+                Pod.get(
+                    client=self.conversion.client,
+                    namespace=self.conversion.namespace,
+                    label_selector=f"conversion={self.conversion.name}",
+                )
+            )
+            raise AssertionError(
+                f"Conversion pods still present {DI_POD_CLEANUP_TIMEOUT}s after cancel: {[p.name for p in remaining]}"
+            )
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)